### PR TITLE
main prototype of card component completed, ui is still not done

### PR DESCRIPTION
**Pull request by JaeBrian**
so the card hopefully all works when creating and deleting one, also the delete icon and adding new post icon need some work, i dont know what will make them look nice or where to put them, almost lost my mind creating card-component, the green circle icons need work - they need to be gray until completed, and maybe we work on the urgency algorithm later...

### CODE REVIEW

File: client/src/pages/HomePage.tsx [hunk 1]
```diff
@@ -1,41 +1,32 @@
...
+import { Post } from '@/api-client/homepageutility';
 import NewLittleThingForm from './NewLittleThingForm';
-import { Button } from '@/components/ui/button';
-import { TrashIcon } from '@radix-ui/react-icons';
+
+import TestComponents from './TestComponents';
 
 const HomePage = () => {
   const [posts, setPosts] = useState<Post[]>([]);
   const { user } = useAuthContext();
```
Comment (sung-kang) [inline client/src/pages/HomePage.tsx]: do we need user context?

File: client/src/pages/HomePage.tsx [hunk 2]
```diff
@@ -1,41 +1,32 @@
...
-  const [deletable, setDeletable] = useState(false);
   const [create, setCreate] = useState(false);
```
Comment (sung-kang) [inline client/src/pages/HomePage.tsx]: i dont think we need state for this and handlers changing this state.
see NewLittleThingsForm.tsx for ref

File: client/src/pages/HomePage.tsx [hunk 3]
```diff
@@ -45,57 +36,37 @@ const HomePage = () => {
...
           <NewLittleThingForm
             handleOpen={handleOpen}
             handleClose={handleClose}
             setPosts={setPosts}
           />
         </div>
+        <div className="flex justify-center items-center">
+          <div className=" grid grid-cols-2 md:grid-cols-2 lg:grid-cols-3 gap-10 p-10 ">
+            {posts.map((post) => (
+              <TestComponents
```
Comment (sung-kang) [inline client/src/pages/HomePage.tsx]: 1. rename component
2. needs key

File: client/src/pages/NewLittleThingForm.tsx
Comment (sung-kang) [inline client/src/pages/NewLittleThingForm.tsx:1]: 1. should be in components directory
2. import DialogClose from dialog and wrap the "Save changes" button with <DialogClose asChild> </DialogClose>. dont need [create, setCreate] state. ie: `<DialogClose asChild> <Button ...>Save changes</Button> </DialogClose>`
3. maybe placeholder values for each label. ie. implying what "littlething" is or "occurrence" means
4. there is typo. "occurence" should be "occurrence"
5. need dropdown or similar for "frequency" as enums of allowed values
6. form needs error handling and/or user feedback on error. maybe toast displaying backend error message or similar

File: client/src/pages/TestComponents.tsx
Comment (sung-kang) [inline client/src/pages/TestComponents.tsx:1]: 1. should be in components directory

File: client/src/pages/TestComponents.tsx [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+          <ul
+            role="list"
+            className="divide-y divide-gray-200 dark:divide-gray-700"
+          >
+            <li className="py-3 sm:py-4">
+              <div className="flex items-center">
+                <div className="flex-shrink-0"></div>
+                <div className="flex-1 min-w-0 ms-4">
+                  <div className="flex-shrink-0"></div>
+                  <p className="text-sm font-medium text-gray-900 truncate dark:text-white">
```
Comment (sung-kang) [inline client/src/pages/TestComponents.tsx]: change to div

File: client/src/pages/TestComponents.tsx [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+                            <div className="flex items-center pt-2">
+                              <span className="text-xs text-muted-foreground">
+                                About Your LittleThing{' '}
+                              </span>
+                            </div>
+                          </div>
+                        </div>
+                      </HoverCardContent>
+                    </HoverCard>
+                  </p>
```
Comment (sung-kang) [inline client/src/pages/TestComponents.tsx]: closing div

File: client/src/pages/TestComponents.tsx [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+  description,
+  frequency,
+  occurence,
+  createdAt,
+  postId,
+  deletable,
+}: CardComponentProps) => {
+  const deletePostHandler = async (postId: string) => {
+    const success = await littlethingsApi.deletePost(postId);
+    if (success) {
```
Comment (sung-kang) [inline client/src/pages/TestComponents.tsx]: needs error handling/user feedback on error

Comment (JaeBrian) [reply client/src/pages/TestComponents.tsx]: will do so later, user will go through confirm delete pop up and error will be displayed there

**Review by sung-kang — APPROVED** [review]
thank you

**Pull request merged**